src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -1022,6 +1024,552 @@ describe('ReactDOMServerIntegration', () => {
     });
   });
 
+  describe('form controls', function() {
+    describe('inputs', function() {
+      itRenders('an input with a value and an onChange', async render => {
+        const e = await render(<input value="foo" onChange={() => {}} />);
+        expect(e.value).toBe('foo');
+      });
+
+      itRenders('an input with a value and readOnly', async render => {
+        const e = await render(<input value="foo" readOnly={true} />);
+        expect(e.value).toBe('foo');
+      });
+
+      itRenders(
+        'an input with a value and no onChange/readOnly',
+        async render => {
+          // this configuration should raise a dev warning that value without
```

Comment:
I suppose it's not clear to me whether we should try to keep all the warnings in sync between the renderers. It seems not-super-valuable if you always revive on the client side, though I suppose if we want to support server-only rendering then we should have all the warnings there too. In any event, this seems good for now.